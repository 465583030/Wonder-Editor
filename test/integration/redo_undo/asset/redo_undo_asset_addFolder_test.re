open Wonder_jest;

open Expect;

open Expect.Operators;

open Sinon;

let _ =
  describe("redo_undo: add folder", () => {
    let sandbox = getSandboxDefaultVal();

    let _simulateAddFolderTwice = () => {
      let assetTreeData =
        MainEditorAssetTreeTool.BuildAssetTree.Folder.TwoLayer.buildOneFolderAssetTree();

      MainEditorAssetHeaderOperateNodeTool.addFolder();
      MainEditorAssetHeaderOperateNodeTool.addFolder();
    };

    let _beforeEach = () => {
      MainEditorSceneTool.initState(~sandbox, ());
      MainEditorSceneTool.createDefaultScene(
        sandbox,
        MainEditorAssetTool.initAssetTree,
      );

      EventListenerTool.buildFakeDom()
      |> EventListenerTool.stubGetElementByIdReturnFakeDom;
    };

    let _afterEach = () => restoreSandbox(refJsObjToSandbox(sandbox^));

    beforeEach(() => sandbox := createSandbox());
    afterEach(() => restoreSandbox(refJsObjToSandbox(sandbox^)));

    RedoUndoTool.testRedoUndoTwoStep(
      sandbox,
      "prepare first step: set currentSceneTreeNode",
      (_simulateAddFolderTwice, _beforeEach, _afterEach),
      BuildComponentTool.buildAssetComponent,
    );
  });