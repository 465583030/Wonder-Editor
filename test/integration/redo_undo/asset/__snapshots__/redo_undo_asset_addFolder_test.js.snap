// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`redo_undo: add folder prepare first step: set currentSceneTreeNode test redo operate test redo one step undo step which from second to zero, redo step which from zero to first 1`] = `
<article
  className="wonder-asset-component"
>
  <div
    className="asset-tree"
  >
    <article
      className="wonder-asset-header"
    >
      <div
        className="asset-header-item"
        onClick={[Function]}
      >
        <div
          className="item-canBeClick"
        >
          <img
            src="./public/img/add.png"
          />
        </div>
      </div>
      <div
        className="asset-header-item"
        onClick={[Function]}
      >
        <div
          className="item-notBeClick"
        >
          <img
            src="./public/img/notRemove.png"
          />
        </div>
      </div>
      <div
        className="asset-header-item"
      >
        <div
          className="item-canBeClick"
        >
          <img
            src="./public/img/load.png"
          />
          <input
            className="asset-fileLoad"
            multiple={true}
            onChange={[Function]}
            type="file"
          />
        </div>
      </div>
    </article>
    <article
      className="wonder-asset-assetTree"
    >
      <ul
        className="wonder-tree-node"
      >
        <li
          style={
            Object {
              "background": "rgba(255,255,255,0.2)",
            }
          }
        >
          <div
            className="item-ground"
            draggable={true}
            onClick={[Function]}
            onDragEnter={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDrop={[Function]}
          />
          <div
            className="item-triangle"
            onClick={[Function]}
          >
            <img
              src="./public/img/down.png"
            />
          </div>
          <img
            className="treeNode-icon"
            src="./public/img/package.png"
          />
          Assets
        </li>
        <ul
          className="wonder-tree-node"
          draggable={true}
          onDragEnd={[Function]}
          onDragStart={[Function]}
        >
          <li
            style={
              Object {
                "border": "0px",
              }
            }
          >
            <div
              className="item-ground"
              draggable={true}
              onClick={[Function]}
              onDragEnter={[Function]}
              onDragLeave={[Function]}
              onDragOver={[Function]}
              onDrop={[Function]}
            />
            <div
              className="item-triangle"
            />
            <img
              className="treeNode-icon"
              src="./public/img/package.png"
            />
            newFolder
          </li>
        </ul>
        <ul
          className="wonder-tree-node"
          draggable={true}
          onDragEnd={[Function]}
          onDragStart={[Function]}
        >
          <li
            style={
              Object {
                "border": "0px",
              }
            }
          >
            <div
              className="item-ground"
              draggable={true}
              onClick={[Function]}
              onDragEnter={[Function]}
              onDragLeave={[Function]}
              onDragOver={[Function]}
              onDrop={[Function]}
            />
            <div
              className="item-triangle"
            />
            <img
              className="treeNode-icon"
              src="./public/img/package.png"
            />
            newFolder 1
          </li>
        </ul>
      </ul>
    </article>
  </div>
  <article
    className="wonder-asset-assetChildren"
  >
    <article
      className="wonder-asset-folderBox"
      id="folder-3"
      style={
        Object {
          "border": "0px",
        }
      }
    >
      <div
        className="item-ground"
        draggable={true}
        onDragEnd={[Function]}
        onDragEnter={[Function]}
        onDragLeave={[Function]}
        onDragOver={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
      />
      <div
        className="box-image"
      >
        <img
          src="./public/img/assetPackage.png"
        />
      </div>
      <div
        className="item-text "
      >
        <span>
          newFolder
        </span>
      </div>
    </article>
    <article
      className="wonder-asset-folderBox"
      id="folder-4"
      style={
        Object {
          "border": "0px",
        }
      }
    >
      <div
        className="item-ground"
        draggable={true}
        onDragEnd={[Function]}
        onDragEnter={[Function]}
        onDragLeave={[Function]}
        onDragOver={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
      />
      <div
        className="box-image"
      >
        <img
          src="./public/img/assetPackage.png"
        />
      </div>
      <div
        className="item-text "
      >
        <span>
          newFolder 1
        </span>
      </div>
    </article>
  </article>
</article>
`;

exports[`redo_undo: add folder prepare first step: set currentSceneTreeNode test redo operate test redo two step undo step which from second to zero,redo step which from zero to second 1`] = `
<article
  className="wonder-asset-component"
>
  <div
    className="asset-tree"
  >
    <article
      className="wonder-asset-header"
    >
      <div
        className="asset-header-item"
        onClick={[Function]}
      >
        <div
          className="item-canBeClick"
        >
          <img
            src="./public/img/add.png"
          />
        </div>
      </div>
      <div
        className="asset-header-item"
        onClick={[Function]}
      >
        <div
          className="item-notBeClick"
        >
          <img
            src="./public/img/notRemove.png"
          />
        </div>
      </div>
      <div
        className="asset-header-item"
      >
        <div
          className="item-canBeClick"
        >
          <img
            src="./public/img/load.png"
          />
          <input
            className="asset-fileLoad"
            multiple={true}
            onChange={[Function]}
            type="file"
          />
        </div>
      </div>
    </article>
    <article
      className="wonder-asset-assetTree"
    >
      <ul
        className="wonder-tree-node"
      >
        <li
          style={
            Object {
              "background": "rgba(255,255,255,0.2)",
            }
          }
        >
          <div
            className="item-ground"
            draggable={true}
            onClick={[Function]}
            onDragEnter={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDrop={[Function]}
          />
          <div
            className="item-triangle"
            onClick={[Function]}
          >
            <img
              src="./public/img/down.png"
            />
          </div>
          <img
            className="treeNode-icon"
            src="./public/img/package.png"
          />
          Assets
        </li>
        <ul
          className="wonder-tree-node"
          draggable={true}
          onDragEnd={[Function]}
          onDragStart={[Function]}
        >
          <li
            style={
              Object {
                "border": "0px",
              }
            }
          >
            <div
              className="item-ground"
              draggable={true}
              onClick={[Function]}
              onDragEnter={[Function]}
              onDragLeave={[Function]}
              onDragOver={[Function]}
              onDrop={[Function]}
            />
            <div
              className="item-triangle"
            />
            <img
              className="treeNode-icon"
              src="./public/img/package.png"
            />
            newFolder
          </li>
        </ul>
        <ul
          className="wonder-tree-node"
          draggable={true}
          onDragEnd={[Function]}
          onDragStart={[Function]}
        >
          <li
            style={
              Object {
                "border": "0px",
              }
            }
          >
            <div
              className="item-ground"
              draggable={true}
              onClick={[Function]}
              onDragEnter={[Function]}
              onDragLeave={[Function]}
              onDragOver={[Function]}
              onDrop={[Function]}
            />
            <div
              className="item-triangle"
            />
            <img
              className="treeNode-icon"
              src="./public/img/package.png"
            />
            newFolder 1
          </li>
        </ul>
        <ul
          className="wonder-tree-node"
          draggable={true}
          onDragEnd={[Function]}
          onDragStart={[Function]}
        >
          <li
            style={
              Object {
                "border": "0px",
              }
            }
          >
            <div
              className="item-ground"
              draggable={true}
              onClick={[Function]}
              onDragEnter={[Function]}
              onDragLeave={[Function]}
              onDragOver={[Function]}
              onDrop={[Function]}
            />
            <div
              className="item-triangle"
            />
            <img
              className="treeNode-icon"
              src="./public/img/package.png"
            />
            newFolder 2
          </li>
        </ul>
      </ul>
    </article>
  </div>
  <article
    className="wonder-asset-assetChildren"
  >
    <article
      className="wonder-asset-folderBox"
      id="folder-3"
      style={
        Object {
          "border": "0px",
        }
      }
    >
      <div
        className="item-ground"
        draggable={true}
        onDragEnd={[Function]}
        onDragEnter={[Function]}
        onDragLeave={[Function]}
        onDragOver={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
      />
      <div
        className="box-image"
      >
        <img
          src="./public/img/assetPackage.png"
        />
      </div>
      <div
        className="item-text "
      >
        <span>
          newFolder
        </span>
      </div>
    </article>
    <article
      className="wonder-asset-folderBox"
      id="folder-4"
      style={
        Object {
          "border": "0px",
        }
      }
    >
      <div
        className="item-ground"
        draggable={true}
        onDragEnd={[Function]}
        onDragEnter={[Function]}
        onDragLeave={[Function]}
        onDragOver={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
      />
      <div
        className="box-image"
      >
        <img
          src="./public/img/assetPackage.png"
        />
      </div>
      <div
        className="item-text "
      >
        <span>
          newFolder 1
        </span>
      </div>
    </article>
    <article
      className="wonder-asset-folderBox"
      id="folder-5"
      style={
        Object {
          "border": "0px",
        }
      }
    >
      <div
        className="item-ground"
        draggable={true}
        onDragEnd={[Function]}
        onDragEnter={[Function]}
        onDragLeave={[Function]}
        onDragOver={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
      />
      <div
        className="box-image"
      >
        <img
          src="./public/img/assetPackage.png"
        />
      </div>
      <div
        className="item-text "
      >
        <span>
          newFolder 2
        </span>
      </div>
    </article>
  </article>
</article>
`;

exports[`redo_undo: add folder prepare first step: set currentSceneTreeNode test undo operate test not undo 1`] = `
<article
  className="wonder-asset-component"
>
  <div
    className="asset-tree"
  >
    <article
      className="wonder-asset-header"
    >
      <div
        className="asset-header-item"
        onClick={[Function]}
      >
        <div
          className="item-canBeClick"
        >
          <img
            src="./public/img/add.png"
          />
        </div>
      </div>
      <div
        className="asset-header-item"
        onClick={[Function]}
      >
        <div
          className="item-notBeClick"
        >
          <img
            src="./public/img/notRemove.png"
          />
        </div>
      </div>
      <div
        className="asset-header-item"
      >
        <div
          className="item-canBeClick"
        >
          <img
            src="./public/img/load.png"
          />
          <input
            className="asset-fileLoad"
            multiple={true}
            onChange={[Function]}
            type="file"
          />
        </div>
      </div>
    </article>
    <article
      className="wonder-asset-assetTree"
    >
      <ul
        className="wonder-tree-node"
      >
        <li
          style={
            Object {
              "background": "rgba(255,255,255,0.2)",
            }
          }
        >
          <div
            className="item-ground"
            draggable={true}
            onClick={[Function]}
            onDragEnter={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDrop={[Function]}
          />
          <div
            className="item-triangle"
            onClick={[Function]}
          >
            <img
              src="./public/img/down.png"
            />
          </div>
          <img
            className="treeNode-icon"
            src="./public/img/package.png"
          />
          Assets
        </li>
        <ul
          className="wonder-tree-node"
          draggable={true}
          onDragEnd={[Function]}
          onDragStart={[Function]}
        >
          <li
            style={
              Object {
                "border": "0px",
              }
            }
          >
            <div
              className="item-ground"
              draggable={true}
              onClick={[Function]}
              onDragEnter={[Function]}
              onDragLeave={[Function]}
              onDragOver={[Function]}
              onDrop={[Function]}
            />
            <div
              className="item-triangle"
            />
            <img
              className="treeNode-icon"
              src="./public/img/package.png"
            />
            newFolder
          </li>
        </ul>
        <ul
          className="wonder-tree-node"
          draggable={true}
          onDragEnd={[Function]}
          onDragStart={[Function]}
        >
          <li
            style={
              Object {
                "border": "0px",
              }
            }
          >
            <div
              className="item-ground"
              draggable={true}
              onClick={[Function]}
              onDragEnter={[Function]}
              onDragLeave={[Function]}
              onDragOver={[Function]}
              onDrop={[Function]}
            />
            <div
              className="item-triangle"
            />
            <img
              className="treeNode-icon"
              src="./public/img/package.png"
            />
            newFolder 1
          </li>
        </ul>
        <ul
          className="wonder-tree-node"
          draggable={true}
          onDragEnd={[Function]}
          onDragStart={[Function]}
        >
          <li
            style={
              Object {
                "border": "0px",
              }
            }
          >
            <div
              className="item-ground"
              draggable={true}
              onClick={[Function]}
              onDragEnter={[Function]}
              onDragLeave={[Function]}
              onDragOver={[Function]}
              onDrop={[Function]}
            />
            <div
              className="item-triangle"
            />
            <img
              className="treeNode-icon"
              src="./public/img/package.png"
            />
            newFolder 2
          </li>
        </ul>
      </ul>
    </article>
  </div>
  <article
    className="wonder-asset-assetChildren"
  >
    <article
      className="wonder-asset-folderBox"
      id="folder-3"
      style={
        Object {
          "border": "0px",
        }
      }
    >
      <div
        className="item-ground"
        draggable={true}
        onDragEnd={[Function]}
        onDragEnter={[Function]}
        onDragLeave={[Function]}
        onDragOver={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
      />
      <div
        className="box-image"
      >
        <img
          src="./public/img/assetPackage.png"
        />
      </div>
      <div
        className="item-text "
      >
        <span>
          newFolder
        </span>
      </div>
    </article>
    <article
      className="wonder-asset-folderBox"
      id="folder-4"
      style={
        Object {
          "border": "0px",
        }
      }
    >
      <div
        className="item-ground"
        draggable={true}
        onDragEnd={[Function]}
        onDragEnter={[Function]}
        onDragLeave={[Function]}
        onDragOver={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
      />
      <div
        className="box-image"
      >
        <img
          src="./public/img/assetPackage.png"
        />
      </div>
      <div
        className="item-text "
      >
        <span>
          newFolder 1
        </span>
      </div>
    </article>
    <article
      className="wonder-asset-folderBox"
      id="folder-5"
      style={
        Object {
          "border": "0px",
        }
      }
    >
      <div
        className="item-ground"
        draggable={true}
        onDragEnd={[Function]}
        onDragEnter={[Function]}
        onDragLeave={[Function]}
        onDragOver={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
      />
      <div
        className="box-image"
      >
        <img
          src="./public/img/assetPackage.png"
        />
      </div>
      <div
        className="item-text "
      >
        <span>
          newFolder 2
        </span>
      </div>
    </article>
  </article>
</article>
`;

exports[`redo_undo: add folder prepare first step: set currentSceneTreeNode test undo operate test undo one step step which from second to first 1`] = `
<article
  className="wonder-asset-component"
>
  <div
    className="asset-tree"
  >
    <article
      className="wonder-asset-header"
    >
      <div
        className="asset-header-item"
        onClick={[Function]}
      >
        <div
          className="item-canBeClick"
        >
          <img
            src="./public/img/add.png"
          />
        </div>
      </div>
      <div
        className="asset-header-item"
        onClick={[Function]}
      >
        <div
          className="item-notBeClick"
        >
          <img
            src="./public/img/notRemove.png"
          />
        </div>
      </div>
      <div
        className="asset-header-item"
      >
        <div
          className="item-canBeClick"
        >
          <img
            src="./public/img/load.png"
          />
          <input
            className="asset-fileLoad"
            multiple={true}
            onChange={[Function]}
            type="file"
          />
        </div>
      </div>
    </article>
    <article
      className="wonder-asset-assetTree"
    >
      <ul
        className="wonder-tree-node"
      >
        <li
          style={
            Object {
              "background": "rgba(255,255,255,0.2)",
            }
          }
        >
          <div
            className="item-ground"
            draggable={true}
            onClick={[Function]}
            onDragEnter={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDrop={[Function]}
          />
          <div
            className="item-triangle"
            onClick={[Function]}
          >
            <img
              src="./public/img/down.png"
            />
          </div>
          <img
            className="treeNode-icon"
            src="./public/img/package.png"
          />
          Assets
        </li>
        <ul
          className="wonder-tree-node"
          draggable={true}
          onDragEnd={[Function]}
          onDragStart={[Function]}
        >
          <li
            style={
              Object {
                "border": "0px",
              }
            }
          >
            <div
              className="item-ground"
              draggable={true}
              onClick={[Function]}
              onDragEnter={[Function]}
              onDragLeave={[Function]}
              onDragOver={[Function]}
              onDrop={[Function]}
            />
            <div
              className="item-triangle"
            />
            <img
              className="treeNode-icon"
              src="./public/img/package.png"
            />
            newFolder
          </li>
        </ul>
        <ul
          className="wonder-tree-node"
          draggable={true}
          onDragEnd={[Function]}
          onDragStart={[Function]}
        >
          <li
            style={
              Object {
                "border": "0px",
              }
            }
          >
            <div
              className="item-ground"
              draggable={true}
              onClick={[Function]}
              onDragEnter={[Function]}
              onDragLeave={[Function]}
              onDragOver={[Function]}
              onDrop={[Function]}
            />
            <div
              className="item-triangle"
            />
            <img
              className="treeNode-icon"
              src="./public/img/package.png"
            />
            newFolder 1
          </li>
        </ul>
      </ul>
    </article>
  </div>
  <article
    className="wonder-asset-assetChildren"
  >
    <article
      className="wonder-asset-folderBox"
      id="folder-3"
      style={
        Object {
          "border": "0px",
        }
      }
    >
      <div
        className="item-ground"
        draggable={true}
        onDragEnd={[Function]}
        onDragEnter={[Function]}
        onDragLeave={[Function]}
        onDragOver={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
      />
      <div
        className="box-image"
      >
        <img
          src="./public/img/assetPackage.png"
        />
      </div>
      <div
        className="item-text "
      >
        <span>
          newFolder
        </span>
      </div>
    </article>
    <article
      className="wonder-asset-folderBox"
      id="folder-4"
      style={
        Object {
          "border": "0px",
        }
      }
    >
      <div
        className="item-ground"
        draggable={true}
        onDragEnd={[Function]}
        onDragEnter={[Function]}
        onDragLeave={[Function]}
        onDragOver={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
      />
      <div
        className="box-image"
      >
        <img
          src="./public/img/assetPackage.png"
        />
      </div>
      <div
        className="item-text "
      >
        <span>
          newFolder 1
        </span>
      </div>
    </article>
  </article>
</article>
`;

exports[`redo_undo: add folder prepare first step: set currentSceneTreeNode test undo operate test undo two step step which from second to zero 1`] = `
<article
  className="wonder-asset-component"
>
  <div
    className="asset-tree"
  >
    <article
      className="wonder-asset-header"
    >
      <div
        className="asset-header-item"
        onClick={[Function]}
      >
        <div
          className="item-canBeClick"
        >
          <img
            src="./public/img/add.png"
          />
        </div>
      </div>
      <div
        className="asset-header-item"
        onClick={[Function]}
      >
        <div
          className="item-notBeClick"
        >
          <img
            src="./public/img/notRemove.png"
          />
        </div>
      </div>
      <div
        className="asset-header-item"
      >
        <div
          className="item-canBeClick"
        >
          <img
            src="./public/img/load.png"
          />
          <input
            className="asset-fileLoad"
            multiple={true}
            onChange={[Function]}
            type="file"
          />
        </div>
      </div>
    </article>
    <article
      className="wonder-asset-assetTree"
    >
      <ul
        className="wonder-tree-node"
      >
        <li
          style={
            Object {
              "background": "rgba(255,255,255,0.2)",
            }
          }
        >
          <div
            className="item-ground"
            draggable={true}
            onClick={[Function]}
            onDragEnter={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDrop={[Function]}
          />
          <div
            className="item-triangle"
            onClick={[Function]}
          >
            <img
              src="./public/img/down.png"
            />
          </div>
          <img
            className="treeNode-icon"
            src="./public/img/package.png"
          />
          Assets
        </li>
        <ul
          className="wonder-tree-node"
          draggable={true}
          onDragEnd={[Function]}
          onDragStart={[Function]}
        >
          <li
            style={
              Object {
                "border": "0px",
              }
            }
          >
            <div
              className="item-ground"
              draggable={true}
              onClick={[Function]}
              onDragEnter={[Function]}
              onDragLeave={[Function]}
              onDragOver={[Function]}
              onDrop={[Function]}
            />
            <div
              className="item-triangle"
            />
            <img
              className="treeNode-icon"
              src="./public/img/package.png"
            />
            newFolder
          </li>
        </ul>
      </ul>
    </article>
  </div>
  <article
    className="wonder-asset-assetChildren"
  >
    <article
      className="wonder-asset-folderBox"
      id="folder-3"
      style={
        Object {
          "border": "0px",
        }
      }
    >
      <div
        className="item-ground"
        draggable={true}
        onDragEnd={[Function]}
        onDragEnter={[Function]}
        onDragLeave={[Function]}
        onDragOver={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
      />
      <div
        className="box-image"
      >
        <img
          src="./public/img/assetPackage.png"
        />
      </div>
      <div
        className="item-text "
      >
        <span>
          newFolder
        </span>
      </div>
    </article>
  </article>
</article>
`;
